core/src/kvs/ds.rs

```diff
@@ -93,18 +94,12 @@ pub struct Datastore {
 	transaction_timeout: Option<Duration>,
 	// Capabilities for this datastore
 	capabilities: Capabilities,
-	engine_options: EngineOptions,
+	pub(crate) engine_options: EngineOptions,
```

Comment:
Nit, but could we limit the export to the parent module? `pub(super)`
(same for `notification_channel` and `lq_cf_store`)?
Or alternatively exposing them through dedicated methods?

Reply:
I have made it exposed to parent module via super; Reluctant to have a java-esque method call. It probably get's reduced with optimisations though, right?